src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Shake128.cs

```diff
@@ -155,10 +164,95 @@ public void GetCurrentHash(Span<byte> destination)
 
             using (ConcurrencyBlock.Enter(ref _block))
             {
+                CheckReading();
                 _hashProvider.Current(destination);
             }
         }
 
+        /// <summary>
+        /// Retrieves the hash for the data accumulated from prior calls to the <c>AppendData</c> methods without
+        /// resetting the object to its initial state and allowing additional calls to continue retrieving the hash.
+        /// </summary>
+        /// <param name="outputLength">The size of the hash to produce.</param>
+        /// <returns>The computed hash.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="outputLength" /> is negative.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">The object has already been disposed.</exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The platform does not support multiple reads of the hash. <see cref="GetHashAndReset(int)" /> can be used
+        ///   to perform a single operation.
+        /// </exception>
+        public byte[] Read(int outputLength)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(outputLength);
+            CheckDisposed();
+
+            using (ConcurrencyBlock.Enter(ref _block))
+            {
+                byte[] output = new byte[outputLength];
+                _hashProvider.Read(output);
+                _reading = true;
+                return output;
+            }
+        }
+
+        /// <summary>
+        /// Fills the buffer with the hash for the data accumulated from prior calls to the <c>AppendData</c> methods without
+        /// resetting the object to its initial state and allowing additional calls to continue retrieving the hash.
+        /// </summary>
+        /// <param name="destination">The buffer to fill with the hash.</param>
+        /// <exception cref="ObjectDisposedException">The object has already been disposed.</exception>
+        /// <exception cref="PlatformNotSupportedException">
```

Comment:
This could probably also use CryptographicException: An error occurred during the cryptographic operation.  That covers the "no concurrent use" exception, and any weird system error codes.